bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,738 @@
+                    "@text/offline.api2.conf-error-access_denied");
+        }
+    }
+
+    /**
+     * Check if a PROPERTY_LEGACY_THING_UID value was set by the discovery process, and if so, clone the legacy thing's
+     * settings into this thing.
+     */
+    private void updateThingFromLegacy() {
+        if (isInitialized()) {
+            logger.warn("updateThingFromLegacy() was called after handler was initialized.");
+            return;
+        }
+        Map<String, String> properties = thing.getProperties();
+        String legacyThingUID = properties.get(PROPERTY_LEGACY_THING_UID);
+        if (Objects.nonNull(legacyThingUID)) {
+            Thing legacyThing = thingRegistry.get(new ThingUID(legacyThingUID));
+            if (Objects.nonNull(legacyThing)) {
+                BridgeBuilder editBuilder = editThing();
+
+                String location = legacyThing.getLocation();
+                if (Objects.nonNull(location) && !location.isBlank()) {
+                    editBuilder = editBuilder.withLocation(location);
+                }
+
+                Object userName = legacyThing.getConfiguration().get(USER_NAME);
+                if (userName instanceof String) {
+                    Configuration configuration = thing.getConfiguration();
+                    configuration.put(Clip2BridgeConfig.APPLICATION_KEY, userName);
+                    editBuilder = editBuilder.withConfiguration(configuration);
+                }
+
+                Map<String, String> newProperties = new HashMap<>(properties);
+                newProperties.remove(PROPERTY_LEGACY_THING_UID);
+
+                updateThing(editBuilder.withProperties(newProperties).build());
+            }
+        }
+    }
+
+    /**
+     * Execute the mass download of all relevant resource types, and inform all child thing handlers.
+     */
+    private void updateThingsNow() {
+        logger.debug("updateThings()");
+        try {
+            Clip2Bridge bridge = getClip2Bridge();
+            for (ResourceReference reference : MASS_DOWNLOAD_RESOURCE_REFERENCES) {
+                ResourceType resourceType = reference.getType();
+                List<Resource> resourceList = bridge.getResources(reference).getResources();
+                if (resourceType == ResourceType.ZONE) {
+                    // add special 'All Lights' zone to the zone resource list
+                    resourceList.addAll(bridge.getResources(BRIDGE_HOME).getResources());
+                }
+                getThing().getThings().forEach(thing -> {
+                    ThingHandler handler = thing.getHandler();
+                    if (handler instanceof Clip2ThingHandler) {
+                        ((Clip2ThingHandler) handler).onResourcesList(resourceType, resourceList);
+                    }
+                });
+            }
+        } catch (ApiException | AssetNotLoadedException e) {
+            // should never happen as we are already online
```

Comment:
In that case it might be okay to log a warning advising user to create issue if this would happen?

Reply:
Ok. Done.